src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+import { updateTodo, deleteTodo } from '../../api/todos';
+
+interface Props {
+  todo?: Partial<Todo> | null;
+  isProcessed: boolean;
+  onDelete?: () => void;
+  onUpdate: (newTitle: string) => void | Promise<void | Partial<Todo>>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isProcessed,
+  onDelete,
+  onUpdate,
+}) => {
+  const [editing, setEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo?.title);
+
+  const handleToggleChecked = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    idToUpdate: number | undefined,
+  ) => {
+    const { target } = event;
+    const { checked } = target;
+
+    if (idToUpdate) {
+      updateTodo(idToUpdate, { completed: checked });
+    }
+  };
+
+  const handleUpdateTitle = (
+    event: React.KeyboardEvent<HTMLSpanElement>,
+  ) => {
+    const { keyCode } = event;
+
+    const { target } = event;
+    const { innerText } = target as HTMLSpanElement;
+
+    const filteredText = innerText.replace(/(\r||\r)/g, '');
+
+    setNewTitle(filteredText);
+
+    if (keyCode === 13) {
```

Comment:
why 13? magic number

Reply:
done:
and also add handling of onBlur event :)